### DISCUSSION TITLE
Multiple Schemas - ParquetSharp.ParquetException: 'class parquet::ParquetException (message: 'Appending to file not implemented.')'

### DISCUSSION THREAD

**Discussion by dotty333** [G-Research/ParquetSharp#474] [Q&A] [answered]
Hi there,
Is it possible to store multiple parquet schemas in a single file? I originally thought not but came across the ManagedRandomAccessFile class but it threw an exception when I tried to write some preamble before creating the ParquetFileWriter:
ParquetSharp.ParquetException: 'class parquet::ParquetException (message: 'Appending to file not implemented.')'

Is this at all possible? I have one main schema and 4 additional ones and would like to avoid multiple files. 
Thanks,
Irfan

**Comment by marcin-krystianc** [comment, marked as answer]
Hi @irfans333 , I don't that it is possible to store multiple schema in a single parquet file.
If you look at the metadata definition, it defines only a single schema filed -> https://github.com/apache/parquet-format/blame/abb92e61e9756016c08401fef729d2ace187d184/src/main/thrift/parquet.thrift#L1151

**Reply by dotty333** [reply]
Hi,
Noted - thanks for the quick response!
